### PR TITLE
feat: auction view for NFT packs

### PR DESCRIPTION
**Pull request by ViacheslavMylostyvyi**
Task: https://app.asana.com/0/1201360309822272/1201458016134730
Description: Same view for NFT packs and regular NFTs while creating an auctio
Знімок екрана  о 16 44 10
n

### CODE REVIEW

File: js/packages/web/src/components/PackCard/index.tsx [hunk 1]
```diff
@@ -53,6 +55,19 @@ const PackCard = ({
...
   return (
     <Card hoverable className="auction-render-card" bordered={false}>
+      {close && (
+        <Button
+          className="card-close-button"
+          shape="circle"
+          onClick={e => {
+            e.stopPropagation();
+            e.preventDefault();
+            close && close();
```
Comment (zaxozhu) [inline js/packages/web/src/components/PackCard/index.tsx:65]:
```suggestion
            close();
```

File: js/packages/web/src/components/PackCard/index.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ interface Props {
   cardsRedeemed?: number;
   allowedAmountToRedeem?: number;
   artView?: boolean;
+  close?: () => void;
```
Comment (zaxozhu) [inline js/packages/web/src/components/PackCard/index.tsx:18]:
```suggestion
  onClose?: () => void;
```

File: js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  current: SafetyDepositDraft;
+  isSelected?: boolean;
+  onSelect: () => void;
+  onClose?: () => void;
+}
+
+const AuctionItemCard = ({ current, isSelected, onSelect, onClose }: IAuctionItemCard) => {
+  const {packs, vouchers} = useMeta();
+  const masterEdition = current.masterEdition?.pubkey;
+  const voucher = Object.values(vouchers).find(v => v?.info?.master === masterEdition);
```
Comment (zaxozhu) [inline js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx:17]: Let's hide it under the env variable and please test it by removing ENABLE_NFT_PACKS from .env

```ts
const shouldShowPacks = process.env.NEXT_ENABLE_NFT_PACKS === 'true';
if (shouldShowPacks) {
  const voucher = Object.values(vouchers).find(v => v?.info?.master === masterEdition);
  if (voucher) { 
     ... 
  }
}
```

File: js/packages/web/src/views/auctionCreate/artSelector.tsx [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import React, { useMemo, useState } from 'react';
 import { Row, Button, Modal, ButtonProps } from 'antd';
-import { ArtCard } from './../../components/ArtCard';
 import { useUserArts } from '../../hooks';
 import { SafetyDepositDraft } from '../../actions/createAuctionManager';
+import AuctionItemCard from "./AuctionItemCard";
```
Comment (zaxozhu) [inline js/packages/web/src/views/auctionCreate/artSelector.tsx:5]:
```suggestion
import AuctionItemCard from './AuctionItemCard';
```

File: js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    } = pack;
+    return (
+      // use <div> for correct grid rendering
+      <div onClick={onSelect}>
+        <PackCard
+          name={name}
+          voucherMetadata={voucher.info.metadata}
+          authority={authority}
+          allowedAmountToRedeem={allowedAmountToRedeem}
+          close={onClose}
```
Comment (zaxozhu) [inline js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx:31]:
```suggestion
          onClose={onClose}
```

**Review by zaxozhu — CHANGES_REQUESTED** [review]

File: js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { ArtCard } from '../../components/ArtCard';
+import PackCard from "../../components/PackCard";
```
Comment (kurpav) [inline js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx:3]: please fix quotes

File: js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  current: SafetyDepositDraft;
+  isSelected?: boolean;
+  onSelect: () => void;
+  onClose?: () => void;
+}
+
+const AuctionItemCard = ({ current, isSelected, onSelect, onClose }: IAuctionItemCard) => {
+  const { packs, vouchers } = useMeta();
+  const shouldShowPacks = process.env.NEXT_ENABLE_NFT_PACKS === 'true';
+  if (shouldShowPacks) {
```
Comment (zaxozhu) [inline js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx:17]:
```suggestion
  if (!shouldShowPacks) {
    return (
    <ArtCard
      pubkey={current.metadata.pubkey}
      preview={false}
      onClick={onSelect}
      className={isSelected ? 'selected-card art-card-for-selector' : 'not-selected-card art-card-for-selector'}
      onClose={onClose}
    />
  );
  }
```

Comment (zaxozhu) [reply js/packages/web/src/views/auctionCreate/AuctionItemCard.tsx:17]: Let's have the shortest block first

**Review by zaxozhu — APPROVED** [review]

**Review by kurpav — APPROVED** [review]

**Pull request merged**